src/bindings/cpp/tests/testcpp_key.cpp

```diff
@@ -13,6 +13,52 @@
 #include <string>
 #include <vector>
 
+/* check if exceptions are thrown if the underlying c-functions return error codes (-1 or NULL) */
+TEST (key, cErrors)
+{
+	Key k ("user:/key", KEY_VALUE, "testkey", KEY_END);
+	Key x = nullptr;
+
+	EXPECT_THROW (x.addName (""), KeyInvalidName);
+	EXPECT_THROW (x.setName (""), KeyInvalidName);
+	EXPECT_THROW (x.addBaseName (""), KeyInvalidName);
+	EXPECT_THROW (x.delBaseName (), KeyInvalidName);
+	EXPECT_THROW (x.setBaseName (""), KeyInvalidName);
+
+	EXPECT_THROW (x.setMeta ("", ""), KeyException);
+	EXPECT_THROW (x.delMeta (""), KeyException);
+	EXPECT_THROW (k.copyAllMeta (nullptr), KeyException);
+	EXPECT_THROW (k.copyMeta (nullptr, "test"), KeyException);
+	EXPECT_THROW (x.rewindMeta (), KeyException);
+
+	EXPECT_THROW (x.set (""), KeyException);
+	EXPECT_THROW (x.copy (nullptr), KeyException);
+	EXPECT_THROW (x.clear (), KeyException);
+	EXPECT_THROW (x.setCallback (nullptr), KeyException);
+	EXPECT_THROW (x.setString (nullptr), KeyException);
+	EXPECT_THROW (x.getString (), KeyException);
+	EXPECT_THROW (x.getReferenceCounter (), KeyException);
+
+	/* Currently not implemented, because otherwise a Key-object
+	 * would not be settable to a nullptr which breaks some existing tests */
+	/* EXPECT_THROW (x++, KeyException); */
```

Comment:
Why would this already throw after the first increment?

Reply:
Because in this case, `x` is a `nullptr` and therefore the `keyIncRef (Key *)` function in [src/libs/elektra/key.c](https://github.com/flo91/libelektra/blob/FLOSS_T1/src/libs/elektra/key.c) returns the error-value `UINT16_MAX`.

```cpp
uint16_t keyIncRef (Key * key)
{
	if (key == NULL)
	{
		return UINT16_MAX;
	}

	if (key->refs == UINT16_MAX - 1)
	{
		return UINT16_MAX;
	}

	return ++key->refs;
}
```